scripts/snark_work_strategy.py

```diff
@@ -0,0 +1,192 @@
+def format_response(response):
+    pending_work = []
+    for work_bundle_spec in response.json().get("data").get("snarkWorkRange"):
+        work_bundle_spec = work_bundle_spec.get("workBundleSpec")
+        pending_work.append(
+            (work_bundle_spec.get("spec"), work_bundle_spec.get("snarkFee"))
+        )
+    return pending_work
+
+
+# There are two types of pending work: completed and not completed yet.
+# For the not completed work, we give the bid given by the CLI.
+# For the already completed work, we halve the corresponding fee.
+def set_fee(pending_work, bid, work_counter):
+    pending_work_with_fee = []
+    for work, fee in pending_work:
+        if fee == None:
+            print(
+                f"{work_counter}-th work not completed yet; performing with fee = {bid}"
+            )
+            pending_work_with_fee.append((work, bid))
+        # some work was already done, do it for half of the original fees
+        elif float(fee) > 0:
+            # fee unit is different when submitting and receiving
+            # 1 submitted = 1000000000 received
+            fee_ratio = 1000000000.0
+            old_fee = float(fee) / fee_ratio
+            new_fee = float(old_fee) / (2.0)
+            print(
+                f"{work_counter}-th work already completed with fee = {old_fee}; performing with fee = {new_fee}"
+            )
+            pending_work_with_fee.append((work, new_fee))
+        else:  # fee = 0, do not perform the work
+            print(f"{work_counter}-th work already completed with fee = {fee}; pass")
+            pass
+        work_counter += 1
+    return pending_work_with_fee
+
+
+# This function returns a range of pending work specified by [start_index] (included) & [end_index] (excluded) obtain via the graphQL endpoint
+def fetch_pending_work(start_index, end_index):
+    body = pending_work_query(start_index, end_index)
+    print(
+        f"Fetching pending work between {start_index} and {end_index} in the pending snark work queue..."
+    )
+    response = requests.post(url=url, json={"query": body})
+    if response.status_code != 200:
+        raise ValueError("PendingSnarkWorkRange graphQL request status code is not 200")
+    return format_response(response)
+
+
+# call [_build/default/src/lib/snark_worker/standalone/run_snark_worker.exe]
+# perform the work given by [work_spec] and send it to the graphql [endpoint]
+def spawn_worker(work_spec, bid, public_key):
+    try:
+        # Create a temporary file to store the work_spec content which may be too large for the command line
+        with tempfile.NamedTemporaryFile(
+            delete=False, mode="w+", encoding="utf-8"
+        ) as temp_file:
+            temp_file.write(f"{work_spec}")
+            temp_file.flush()  # Ensure the content is written to disk
+            temp_file.seek(0)
+            cmd = [
+                    "_build/default/src/lib/snark_worker/standalone/run_snark_worker.exe",
```

Comment:
I would propose to introduce const for this app and place it on top of this script with meaningful comment that this script requires run_snark_worker app being already built

Reply:
Done